arbos/block_processor.go

```diff
@@ -240,6 +240,12 @@ func ProduceBlockAdvanced(
 					options = hooks.ConditionalOptionsForTx[0]
 					hooks.ConditionalOptionsForTx = hooks.ConditionalOptionsForTx[1:]
 				}
+			} else {
+				state, err = arbosState.OpenSystemArbosState(statedb, nil, true)
+				if err != nil {
+					return nil, nil, err
+				}
+				header = createNewHeader(lastBlockHeader, l1Info, state, chainConfig)
```

Comment:
Currently, in this function, Extra and MixData are just the last header's values. We should recreate them, as we do in IIRC FinalizeBlock

Reply:
done.